### DISCUSSION TITLE
How to mark a single file module as py.typed?

### DISCUSSION THREAD

**Discussion by RobertCraigie** [microsoft/pyright#2910] [Q&A] [answered]
I'm using https://github.com/python-distro/distro which is a single file module: `distro.py` which provides type hints in the form of type comments.

However pyright reports `error: Stub file not found for "distro" (reportMissingTypeStubs)` due to the lack of a `py.typed` file.

Would it be possible for the maintainers of `distro` to mark the module as being typed?

**Comment by JelleZijlstra** [comment, marked as answer]
PEP 561 doesn't support marking a single-file distribution as typed: https://www.python.org/dev/peps/pep-0561/#packaging-type-information. The distribution would have to migrate to using a package, or provide a separate stubs package.

We'd be happy to accept stubs for distro over in https://github.com/python/typeshed.
